docs/data/date-pickers/date-picker/HelperText.tsx

```diff
@@ -1,24 +1,18 @@
 import * as React from 'react';
-import { Dayjs } from 'dayjs';
-import TextField from '@mui/material/TextField';
 import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
 import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
-import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import { Unstable_NextDatePicker as NextDatePicker } from '@mui/x-date-pickers/NextDatePicker';
 
 export default function HelperText() {
-  const [value, setValue] = React.useState<Dayjs | null>(null);
-
   return (
     <LocalizationProvider dateAdapter={AdapterDayjs}>
-      <DatePicker
+      <NextDatePicker
         label="Helper text example"
-        value={value}
-        onChange={(newValue) => {
-          setValue(newValue);
+        componentsProps={{
+          input: {
+            helperText: 'MM / DD / YYYY',
```

Comment:
Here we have a small problem

The format is determined inside `useDateField`
But `useDateField` is called with the props resolved from the slots.

The only solution I see to keep this feature possible is to duplicate the format logic (extract https://github.com/mui/mui-x/blob/next/packages/x-date-pickers/src/DateField/useDateField.ts#L25 inside a tiny hook) and call it on `DateField` as well)
Same for the other fields

If you have another solution, I am interested.

Reply:
We already callbacks on the slot (when they are built using `useSlotProps`).
But in the params of your function, you don't have the format yet, because the function is called before `useDateField` which computes the format.

Reply:
Thinking about this case a bit more—I don't think that we have any sort of problem here.
Static props can be used to pass static props, but in case something more advanced is needed with access to real/current props—the custom component can be passed down. 👍

Reply:
I feel like being able to put the format in the helper test is something quite a lot of people want.
And not being able to easily do it (without loosing the automatic format of the pickers / field) is not great.

But we can see if people complain about it, we have solutions if needed

Reply:
> olution could be to let dev compute the format helper themself:
> 
> ```
> adapter.getFormatHelperText(adapter.formats.keyboardDate)
> ```

Soor @alexfauquette I missed your answer
Sure the way it was done was hacky, and if we want to support it, I'm in favor of passing the format in a `format` property, not a `placeholder` property.

The only issue with your proposal is that the user need to know the format.
He would loose the ability to have the picker decide the format based on the locale / the views / the meridiem etc...

Maybe that's not that big of a deal, we can wait for feedbacks.

Reply:
> Maybe that's not that big of a deal, we can wait for feedbacks.

I'd say that having the option to access current localized `format` in one way or another for different use cases might be the expectation by users—especially ones used to using previously available `placeholder` prop.
Based on this assumption—it would be worthwhile to think of the easiest way to expose it.
Maybe it could be initialized in the components `format` props? 🤔  Now it seems to be empty if user did not provide any format...

Reply:
> Maybe it could be initialized in the components format props? 

Right now the format is initialized in `useDateField` (or the equivalent for other fields)
Which is called after the resolution of the slots because it needs the result from this resolution
But we could either

1. Make the `format` param mandatory in `useDateField` and apply its default value inside the component instead of the hook

2. Apply the default value both in the component and in the hook (more duplication but people using the hook directly could still benefit from it)